### ISSUE TITLE
Umlauts are not displayed correctly in import templates and csv exports

### ISSUE THREAD

**Issue by syntakker** [hzi-braunschweig/SORMAS-Project#4045]
### Bug Description
Import templates generated by SORMAS are ASCII encoded:
```
$ ls | xargs file
sormas_import_area_template.csv:             ASCII text
sormas_import_case_contact_template.csv:     ASCII text, with very long lines
sormas_import_case_template.csv:             ASCII text, with very long lines
sormas_import_community_template.csv:        ASCII text
sormas_import_contact_template.csv:          ASCII text, with very long lines
sormas_import_country_template.csv:          ASCII text
sormas_import_district_template.csv:         ASCII text
sormas_import_eventparticipant_template.csv: ASCII text, with very long lines
sormas_import_facility_template.csv:         ASCII text, with very long lines
sormas_import_line_listing_template.csv:     ASCII text, with very long lines
sormas_import_point_of_entry_template.csv:   ASCII text
sormas_import_population_data_template.csv:  ASCII text, with very long lines
sormas_import_region_template.csv:           ASCII text
```
which means there is no chance to have special characters like umlauts displayed correctly.

CSV exports are equally not correctly displayed when downloaded under Windows and opened in Excel.

### Steps to Reproduce
In the case list, download the import template in the import dialog (with `country.locale=de-DE` in `sormas.properties`).

### Expected Behavior
- [x] Import templates should be generated in UTF-8. (implemented in #4322) 
- [ ] For a seamless workflow, downloads should be ISO-8859 encoded for Windows clients and UTF-8 encoded otherwise (to be discussed, see draft PR  #4046)

### Screenshots
![image](https://user-images.githubusercontent.com/5616564/104791365-bcdb4980-579a-11eb-80a8-202e8f04645b.png)

### System Details
* Device: Local PC
* SORMAS version: 1.54.0-SNAPSHOT
* Android version/Browser: Firefox
* Server URL: localhost
* User Role: admin

**Comment by syntakker** [comment]
see also #3039.

**Comment by Marko-ilmkreis** [comment]
See also #3747

**Comment by syntakker** [comment]
Further analysis:
1. Import templates should indeed be generated in UTF-8, otherwise there is no chance to display any special characters correctly
2. Serving CSV downloads in ISO-8859 for Windows systems would very likely smoothen the workflow SORMAS -> Excel (just download and open in Excel). However it could be contraproductive for regions not using a Latin-1 charset.

Proposal:
1. I will move import template generation in UTF-8 to a PR of its own, so it can be treated separately
2. Serving ISO-8859 for Windows systems can be an optional feature and can be enabled /disabled by configuration

**Comment by Marko-ilmkreis** [comment]
@syntakker is it possible to show a message if the user tries to import a file which is not UTF-8? I think there might be some users, which are not familiar with it and a clear error message would help them.

**Comment by syntakker** [comment]
@kwa20 do you have feedback from the users on generated import templates? The first part - template generation in UTF-8 - should be fixed in #4322. Unclear to me however is, if and how an option to download them in ISO-8859-1 encoding on Windows systems would smoothen the workflow - assuming that CSVs are opened in Excel - and should be offered

If so, options are:
1. determine Client system and serve files accordingly (ISO-8859 for Windows, UTF-8 otherwise)
2. make this configurable in `sormas.properties`
3. offer a radio button to choose (consider however where it should be placed)

**Comment by kwa20** [comment]
@syntakker I don't have user feedback yet because systems have not been updated thus far. I can however confirm that import templates are generated with placeholders for umlauts. What about generating them in UTF-8 with BOM when the generated template contains umlauts? This is also now accepted as an encoding for an import.
<img width="157" alt="templatesUmlauts" src="https://user-images.githubusercontent.com/70317594/107479191-1016a100-6b7b-11eb-9060-b73c4917312e.PNG">

**Comment by syntakker** [comment]
@kwa20 this issue is still in discussion, so it is not deployed anywhere. It is rather about determining the best solution.

Which system you are testing on? You're screenshot looks as if it is at least UTF-8. On my local system, the templates are generated as ASCII files, where special characters display as "?".

Here's the import template in different encodings (please rename to ".csv", github does not support ".csv" uploads):

[sormas_import_case_template_UTF8.txt](https://github.com/hzi-braunschweig/SORMAS-Project/files/5957476/sormas_import_case_template_UTF8.txt)
[sormas_import_case_template_UTF8_BOM.txt](https://github.com/hzi-braunschweig/SORMAS-Project/files/5957475/sormas_import_case_template_UTF8_BOM.txt)
[sormas_import_case_template_ISO_5589_1.txt](https://github.com/hzi-braunschweig/SORMAS-Project/files/5957473/sormas_import_case_template_ISO_5589_1.txt)

- Can you test which of these work for you in Excel?
- Are you aware of other tools or tool chains users use besides Excel (and maybe LibreOffice)?

**Comment by syntakker** [comment]
@kwa btw, I've seen the "?" popping up in GA demos, too. I suppose this was on demoversion.sormas-oegd.de/sormas-ui.

**Comment by kwa** [comment]
> @kwa btw, I've seen the "?" popping up in GA demos, too. I suppose this was on demoversion.sormas-oegd.de/sormas-ui.

@syntakker you sent your message to the wrong user :) -> @kwa20 is the one you want I think :)

**Comment by syntakker** [comment]
> > @kwa btw, I've seen the "?" popping up in GA demos, too. I suppose this was on demoversion.sormas-oegd.de/sormas-ui.
> 
> @syntakker you sent your message to the wrong user :) -> @kwa20 is the one you want I think :)

@kwa that's totally true, thank you. Nice to meet you anyway :-)

**Comment by kwa20** [comment]
@syntakker I just tried this again on the demoversion.sormas-oegd.de (1.59.2) and both, import templates and exports are encoded as UTF-8 for me. Only checked this for cases.

**Comment by syntakker** [comment]
@kwa20 it's an old issue anyways, it just popped up again because I confused your user name a while ago :-)
I think we can close this, the encoding problems have been solved in #4322 and #3747.
